src/providers/arcgisrest/qgsafsprovider.cpp

```diff
@@ -653,6 +653,9 @@ bool QgsAfsProvider::setSubsetString( const QString &subset, bool )
   if ( trimmedSubset == mSharedData->subsetString() )
     return true;
 
+  // We must not change the subset string of the shared data used in another iterator/data provider ...
```

Comment:
Can we add a bit more to the comment to explain the behavior / reasoning a bit more?

Reply:
done